great_expectations/datasource/data_connector/inferred_asset_azure_data_connector.py

```diff
@@ -0,0 +1,173 @@
+                ).group(1)
+                self._azure = BlobServiceClient.from_connection_string(**azure_options)
+            elif account_url is not None:
+                self._account_name = re.search(
+                    r"(?:https?://)?(.+?).blob.core.windows.net", account_url
+                ).group(1)
+                self._azure = BlobServiceClient(**azure_options)
+        except (TypeError, AttributeError):
+            raise ImportError(
+                "Unable to load Azure BlobServiceClient (it is required for InferredAssetAzureDataConnector). \
+                Please ensure that you have provided the appropriate keys to `azure_options` for authentication."
+            )
+
+    def build_batch_spec(self, batch_definition: BatchDefinition) -> AzureBatchSpec:
+        """
+        Build BatchSpec from batch_definition by calling DataConnector's build_batch_spec function.
+
+        Args:
+            batch_definition (BatchDefinition): to be used to build batch_spec
+
+        Returns:
+            BatchSpec built from batch_definition
+        """
+        batch_spec: PathBatchSpec = super().build_batch_spec(
+            batch_definition=batch_definition
+        )
+        return AzureBatchSpec(batch_spec)
+
+    def _get_data_reference_list(
+        self, data_asset_name: Optional[str] = None
+    ) -> List[str]:
+        """
+        List objects in the underlying data store to create a list of data_references.
+
+        This method is used to refresh the cache.
+        """
+        query_options: dict = {
+            "container": self._container,
+            "name_starts_with": self._name_starts_with,
+            "delimiter": self._delimiter,
+        }
+
+        path_list: List[str] = list_azure_keys(
+            azure=self._azure,
+            query_options=query_options,
+            recursive=True,
+        )
+        return path_list
+
+    def _get_full_file_path(
+        self,
+        path: str,
+        data_asset_name: Optional[str] = None,
+    ) -> str:
+        # data_asset_name isn't used in this method.
+        # It's only kept for compatibility with parent methods.
+        # Pandas and Spark execution engines utilize separate path formats for accessing Azure Blob Storage service.
+        full_path: str
+        if isinstance(self.execution_engine, PandasExecutionEngine):
+            full_path = os.path.join(
+                f"{self._account_name}.blob.core.windows.net", self._container, path
+            )
+        elif isinstance(self.execution_engine, SparkDFExecutionEngine):
+            full_path = os.path.join(
+                f"{self._container}@{self._account_name}.blob.core.windows.net", path
+            )
+            full_path = f"wasbs://{full_path}"
+        else:
+            raise ge_exceptions.DataConnectorError(
+                f"""Illegal ExecutionEngine type "{str(type(self.execution_engine))}" used in \
+"{self.__class__.__name__}".
+"""
+            )
```

Comment:
love it! Can we ensure that we have unit tests for each path?

Reply:
@cdkini They are in `tests/core/test_util.py`.

Reply:
That's for `AzureUrl` no? I see a full_file_path test as the last unit test in the test_inferred_asset file so I figure it probably goes there?

Reply:
Three new tests (with a comment to watch out for the future work) have been added.  Thanks!